### PR TITLE
Moe 2403

### PR DESCRIPTION
**Pull request by Moetez16**
Don't merge after Approve

### CODE REVIEW

**Comment by Moetez16** [comment]
Don't merge!!!!!

File: app/grails-app/controllers/com/k_int/kbplus/SurveyController.groovy [hunk 1]
```diff
@@ -26,6 +28,8 @@ import java.text.DateFormat
 import java.text.NumberFormat
 import java.text.SimpleDateFormat
 
+import static de.laser.helper.RDStore.getTIPP_STATUS_DELETED
```
Comment (klodav) [inline app/grails-app/controllers/com/k_int/kbplus/SurveyController.groovy:31]: DONT

Comment (croqueGrec09) [reply app/grails-app/controllers/com/k_int/kbplus/SurveyController.groovy:31]: Could be one of IntelliJ's detections ...

File: app/grails-app/i18n/messages.properties [hunk 2]
```diff
@@ -3751,6 +3751,23 @@ survey.plural = Surveys
...
+copySurvey.copyDocs = Copy docs
+copySurvey.copyAnnouncements = Copy notes
+copySurvey.copyTasks = Copy Tasks
+copySurvey.copyComment = Copy Comment
+copySurvey.copySurveyConfigUrl = Copy information url
+copySurvey.copyParticipants = Copy Participants
+copySurvey.copySurveyConfigComment = Copy spec. Comment
+copySurvey.copyInSelectedSubs = Copy survey into selected licenses
+copySurvey.subscription.info = Select here the subscription you want to copy this survey into.
+copySurvey.subscription.info2 = Copy selected survey properties in the following licenses:
```
Comment (croqueGrec09) [inline app/grails-app/i18n/messages.properties:3770]: subscriptions?

Comment (croqueGrec09) [reply app/grails-app/i18n/messages.properties:3770]: (see also in line 3768)

File: app/grails-app/controllers/com/k_int/kbplus/SurveyController.groovy [hunk 3]
```diff
@@ -4928,4 +5111,96 @@ class SurveyController {
             return false
         }
     }
+    
+    boolean copySurveyConfigCharacteristic(SurveyConfig oldSurveyConfig, SurveyConfig newSurveyConfig, params){
+
+        oldSurveyConfig.documents?.each { dctx ->
+                //Copy Docs
+                if (params.copySurvey.copyDocs) {
+                    if (((dctx.owner?.contentType == 1) || (dctx.owner?.contentType == 3)) && (dctx.status?.value != 'Deleted')) {
```
Comment (klodav) [inline app/grails-app/controllers/com/k_int/kbplus/SurveyController.groovy:5118]: Please use existing constants; Doc.CONTENT_TYPE_<xy>

File: app/grails-app/services/de/laser/FilterService.groovy [hunk 4]
```diff
@@ -370,6 +371,12 @@ class FilterService {
             params.filterSet = true
         }
 
+        if(params.ids) {
+            query << "surInfo.id in :ids"
```
Comment (klodav) [inline app/grails-app/services/de/laser/FilterService.groovy:375]: please use convention: **surInfo.id in (:ids)**

File: app/grails-app/views/survey/copySurvey.gsp [hunk 5]
```diff
@@ -0,0 +1,586 @@
...
+
+                    </tr>
+
+                    <tr>
+                        <g:sortableColumn class="la-smaller-table-head" params="${params}" property="s.endDate"
+                                          title="${message(code: 'default.endDate.label')}"/>
+                    </tr>
+                    </thead>
+                    <g:each in="${subscriptions}" var="s" status="i">
+                        <g:if test="${true || !s.instanceOf}">
```
Comment (klodav) [inline app/grails-app/views/survey/copySurvey.gsp:430]: hehe

File: app/grails-app/views/survey/copySurvey.gsp [hunk 6]
```diff
@@ -0,0 +1,586 @@
...
+                    </table>
+                    </td>
+                </tr>
+
+                <tr>
+                    <th><g:checkBox name="copySurvey.copyDocs" value="${true}"/></th>
+                    <th>${message(code: 'copySurvey.copyDocs')}</th>
+                    <td>
+                        <g:each in="${surveyConfig.documents.sort { it.owner?.title }}" var="docctx">
+                            <g:if test="${(((docctx.owner?.contentType == 1) || (docctx.owner?.contentType == 3)) && (docctx.status?.value != 'Deleted'))}">
```
Comment (klodav) [inline app/grails-app/views/survey/copySurvey.gsp:184]: Please use existing constants; Doc.CONTENT_TYPE_\<xy>

File: app/grails-app/views/survey/copySurvey.gsp (hunk 5 above)
Comment (Moetez16) [reply app/grails-app/views/survey/copySurvey.gsp:430]: -.-

File: app/grails-app/controllers/com/k_int/kbplus/SurveyController.groovy (hunk 3 above)
Comment (croqueGrec09) [reply app/grails-app/controllers/com/k_int/kbplus/SurveyController.groovy:5116]: Is legacy and should be reviewed sidewidely

File: app/grails-app/services/de/laser/FilterService.groovy (hunk 4 above)
Comment (croqueGrec09) [reply app/grails-app/services/de/laser/FilterService.groovy:375]: It works also without

File: app/grails-app/i18n/messages.properties (hunk 2 above)
Comment (croqueGrec09) [reply app/grails-app/i18n/messages.properties:3770]: in the following licenses -> following subscriptions

File: app/grails-app/views/survey/copySurvey.gsp (hunk 6 above)
Comment (croqueGrec09) [reply app/grails-app/views/survey/copySurvey.gsp:184]: What is with this remark?

File: app/grails-app/views/survey/copySurvey.gsp [hunk 7]
```diff
@@ -0,0 +1,586 @@
...
+                                            <g:if test="${s.consortia && s.consortia == institution}">
+                                                ( ${s.subscriber?.name} )
+                                            </g:if>
+                                        </g:if>
+                                    </g:link>
+                                    <g:if test="${s.owner}">
+                                        <div class="la-flexbox">
+                                            <i class="icon balance scale la-list-icon"></i>
+                                            <g:link controller="license" action="show"
+                                                    id="${s.owner.id}">${s.owner?.reference ?: message(code: 'missingLicenseReference', default: '** No License Reference Set **')}</g:link>
```
Comment (croqueGrec09) [inline app/grails-app/views/survey/copySurvey.gsp:456]: If I check already for s.owner.id, s.owner cannot be numm when asking for reference

File: app/grails-app/views/survey/copySurvey.gsp [hunk 8]
```diff
@@ -0,0 +1,586 @@
...
+                                </td>
+                                <td>
+                                <!-- packages -->
+                                    <g:each in="${s.packages.sort { it?.pkg?.name }}" var="sp" status="ind">
+                                        <g:if test="${ind < 10}">
+                                            <div class="la-flexbox">
+                                                <i class="icon gift la-list-icon"></i>
+                                                <g:link controller="subscription" action="index" id="${s.id}"
+                                                        params="[pkgfilter: sp.pkg?.id]"
+                                                        title="${sp.pkg?.contentProvider?.name}">
```
Comment (croqueGrec09) [inline app/grails-app/views/survey/copySurvey.gsp:468]: sp.pkg - to be or not to be?

**Pull request merged**